### ISSUE TITLE
Prediction with two chains/ MSA issue

### ISSUE THREAD

**Issue by jbogomolovas2** [ohuelab/boltzina#9]
Hi, and thank you for the fantastic tool — Boltz/Boltzina have been extremely useful in my work. However I have run into some nuissance recently.

I am using a two-chain protein. The YAML is only used for Boltz-2, and Boltz-2 runs correctly: both chains get valid MSAs, and the output directory (manifest, predictions, MSAs) looks normal.

The issue appears only when running Boltzina for affinity scoring.
Boltzina loads the MSA for the first chain, but the second chain’s MSA (generated by Boltz-2) is always rejected with:

MSA does not match input sequence, creating dummy

If I remove the second chain from the Boltz-2 YAML, Boltzina shows no warning.
If I swap the order of chains in the Boltz-2 YAML, the newly second chain becomes the one whose MSA Boltzina rejects.

So it seems Boltzina cannot read or accept the second chain’s MSA produced by Boltz-2, even though Boltz-2 itself uses it successfully.

Is this a known limitation, or is there a way to make Boltzina load MSAs for both chains?

I can provide the Boltz-2 YAML, sequences, MSAs, manifest.json, and Boltzina logs if needed.

Thank you.

**Comment by jbogomolovas2** [comment]
Thank you for your prompt respose! I will definitely try this out in next 24 hours. However in my situation complex of two proteins is receptor for a ligand. Can I specify two chains as a receptor?

**Comment by YumizSui** [comment]
Thanks for the update.
I checked that the new version works with the other multi-chain case `sample/CDK2_ab`.
Could you please share the console output or error logs? I’d like to know exactly what you mean by "does not recognize MSA".

**Comment by jbogomolovas2** [comment]
Thank you so much for looking into this issue and providing a sample for testing. Indeed. I run your sample with no problems. My complex however throws error oout:(boltzina) julius@julius-Precision-T5610:~/boltzina$ python3 run.py CHIP/boltzina_scoring_test.json
--------------------------------
Output directory: /home/julius/boltzina/CHIP/boltzina_test
Seed: None
Mode: scoring only
Using float32 matmul precision: highest
Running scoring only...
Running scoring-only mode for 1 ligand poses...
Scoring poses with Boltzina...
/home/julius/envs/boltzina/lib/python3.12/site-packages/pytorch_lightning/utilities/migration/utils.py:56: The loaded checkpoint was produced with Lightning v2.5.0.post0, which is newer than your current Lightning version: v2.5.0
Using bfloat16 Automatic Mixed Precision (AMP)
GPU available: True (cuda), used: True
TPU available: False, using: 0 TPU cores
HPU available: False, using: 0 HPUs
/home/julius/envs/boltzina/lib/python3.12/site-packages/pytorch_lightning/trainer/connectors/logger_connector/logger_connector.py:76: Starting from v1.9.0, `tensorboardX` has been removed as a dependency of the `pytorch_lightning` package, due to potential conflicts with other packages in the ML ecosystem. For this reason, `logger=True` will use `CSVLogger` as the default logger, unless the `tensorboard` or `tensorboardX` packages are found. Please `pip install lightning[extra]` or one of them to enable TensorBoard support by default
LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]
Predicting: |                                                                                      | 0/? [00:00<?, ?it/s]Warning: MSA does not match input sequence, creating dummy. 2 [22] [16 18 11  8  8 21  5] CHIP_affinity2_0_1
Warning: MSA does not match input sequence, creating dummy. 2 [22] [16 18 11  8  8 21  5] CHIP_affinity2_0_1
Predicting DataLoader 0: 100%|█████████████████████████████████████████████████████████████| 1/1 [00:20<00:00,  0.05it/s]Number of failed examples: 0
Predicting DataLoader 0: 100%|█████████████████████████████████████████████████████████████| 1/1 [00:20<00:00,  0.05it/s]
Saving results to CSV...
Results saved to /home/julius/boltzina/CHIP/boltzina_test/boltzina_results.csv
   affinity_pred_value  affinity_probability_binary  affinity_pred_value1  ...  ligand_idx  docking_rank  docking_score
0            -0.391825                     0.908857             -1.536764  ...           0             1           None

[1 rows x 10 columns]
It seems to be "me" problem- associated with my dataset, so will investigate my data on my own. Thank you so much
